core/internal/runmetric/runmetric.go

```diff
@@ -200,3 +202,16 @@ func (mh *MetricHandler) matchGlobMetric(key string) (definedMetric, bool) {
 
 	return definedMetric{}, false
 }
+
+func (mh *MetricHandler) splitEscapedDottedMetricName(metricName string) []string {
+	// Match any period character that is not preceded by a backslash.
+	re := regexp.MustCompile(`.*?[^\\](\.|$)`)
+	parts := re.FindAllString(metricName, -1)
```

Comment:
I think for full generality, we may want to consider escaped backslashes as well:

```
a.b => ["a", "b"]
a\.b => ["a.b"]
a\\.b => ["a\", "b"]
a\\\.b => ["a\.b"]
```

I admit that these examples aren't very realistic of course, but there's a straightforward and efficient implementation using a single for-loop and a boolean state variable, which may be even easier to reason about than a regex-based split. Just something to consider!

(regarding that implementation, I'd probably choose to map `"c:\why\would\anyone\do\this\.exe"` to `["c:\why\would\anyone\do\this.exe"]`, though `["c:whywouldanyonedothis.exe"]` would be a reasonable alternative)

Reply:
Good call out, the for loop is easier to work through without any of the regex mysticism.